test/with_dummyserver/test_proxy_poolmanager.py

```diff
@@ -78,6 +78,16 @@ def test_https_proxy(self) -> None:
             r = https.request("GET", f"{self.http_url}/")
             assert r.status == 200
 
+    def test_http_and_https_kwarg_ca_cert_data_proxy(self) -> None:
+        with proxy_from_url(
+            self.https_proxy_url, ca_cert_data=open(DEFAULT_CA, "rb").read()
```

Comment:
I guess the test fails in CI because `DEFAULT_CA` contains a PEM-encoded certificate and it is not provided as a string.

https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_cert_chain
> The cadata object, if present, is either an ASCII string of one or more PEM-encoded certificates or a [bytes-like object](https://docs.python.org/3/glossary.html#term-bytes-like-object) of DER-encoded certificates. Like with capath extra lines around PEM-encoded certificates are ignored but at least one certificate must be present.

Reply:
Please remember to close the file properly, use `with` for opening it

Reply:
Yes, Thanks for the review. Made changes